### ISSUE TITLE
Expose results via outputs

### ISSUE THREAD

**Issue by aknosis** [DataDog/synthetics-ci-github-action#120]
It would be extremely helpful if the data shown in the action's logs could also be set as outputs for accessibility in other parts of a workflow. For example, adding the report as a comment on a PR, extracting links and appending to a PR, external notification of pass/fail status.

I do not know the entire structure of synthetic tests, but based on [this run](https://github.com/DataDog/synthetics-ci-github-action/actions/runs/4204210853/jobs/7294573303#step:4:21) as an example, I would see the following outputs being useful:

- `result: succeeded|failed`
- `resultsUrl: https://app.datadoghq.com/synthetics/explorer/ci?batchResultId=1eba11d9-e0f8-43c0-ba2c-dfce4f4c0dab`
- `criticalErrors: int`
- `passed: int`
- `failedNonBlocking: int`
- `failed: int`
- `skipped: int`
- `notFound: int`
- `timedOut: int`
- `report: string` (the entire string from === REPORT == and below)
- `testRuns: array` (output as a json string)
```json
[
   {
      "passed":2,
      "failed":0,
      "batchUrl":"https://app.datadoghq.com/synthetics/explorer/ci?batchResultId=1eba11d9-e0f8-43c0-ba2c-dfce4f4c0dab",
      "runs":[
         {
            "id":"2r9-q7u-4nn",
            "detailsUrl":"https://app.datadoghq.com/synthetics/details/2r9-q7u-4nn/result/3868113174340701979?from_ci=true",
            "device":"chrome.laptop_large",
            "duration":1204,
            "location":"Paris (AWS)",
            "name":"[DO_NOT_DELETE] Used in `datadog-ci` repo for CI testing",
            "result":"passed|failed"
         }
      ]
   }
]
```

**Comment by Drarig29** [comment]
Hi @aknosis! Sorry for the delay, and thank you for the feature request! 

That's a really good idea, we'll add it in our backlog 😄 
Cheers!

**Comment by spirosikmd** [comment]
Hi team! Is there an update on when this feature will be implemented? Has anyone else found a temporary way of doing this?

**Comment by Drarig29** [comment]
Hi @spirosikmd! What's your use case and which output in particular would you be interested in?

**Comment by florianmutter** [comment]
> Hi [@spirosikmd](https://github.com/spirosikmd)! What's your use case and which output in particular would you be interested in?

Our use case is sending a MS Teams message if the synthetic tests fail. Currently we just send the log output which is rather hard to read.

**Comment by spirosikmd** [comment]
Ideally, we can do something lilke the following. The `BATCH_RESULTS_URL` (just an example) would be set by the `synthetics-ci-github-action`.

```yaml
- name: Run Datadog Synthetic tests
   uses: DataDog/synthetics-ci-github-action@v2.2.0
   id: tests
   with:
     # ...

- name: Post report link
   if: ${{ !cancelled() }}
   uses: GuiBranco/github-status-action-v2@v1.1.13
   with:
     authToken: ${{ secrets.GITHUB_TOKEN }}
     context: 'Synthetic tests / Report'
     state: 'success'
     sha: ${{ github.sha }}
     target_url: ${{ steps.tests.outputs.BATCH_RESULTS_URL }}
```

**Comment by jimjkelly** [comment]
@Drarig29 My company has (or had, we actually parked it due to not being able to get the data out of this, but perhaps we'd pick it up again) a use case around running tests and tagging them with test case tickets that the test validates.  This way we can automatically validate  things like Zephyr tests for a release by taking the output from here, extracting the tickets a synthetic validates, then call the zephyr APIs to mark the result.

**Comment by florianmutter** [comment]
The raw results can easily be too big to be handled by github actions as outputs and it prints an `Argument list too long` error. I believe there should be an option to write the raw result to the file instead of an output
